src/System.Net.Requests/src/System/Net/RequestStream.cs

```diff
@@ -99,16 +99,52 @@ public override void SetLength(long value)
 
         public override void Write(byte[] buffer, int offset, int count)
         {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
+            }
+            if (offset < 0 || offset > buffer.Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(offset));
+            }
+            if (count < 0 || count > (buffer.Length - offset))
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
+            }
             _buffer.Write(buffer, offset, count);
         }
 
         public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
```

Comment:
>Should be returning an errored Task rather than throwing directly?

Not necessarily.  Most argument checking is done directly and not with errors returned into the task.  In general, it will follow the same behavior as .NET Framework.  But the overall guidance has always been that "developer" errors like this will throw exceptions before a Task is generated.